include/evm_runtime/config_wrapper.hpp

```diff
@@ -0,0 +1,128 @@
+#pragma once
+#include <eosio/eosio.hpp>
+#include <evm_runtime/tables.hpp>
+
+namespace evm_runtime {
+
+using eosio::check;
+
+struct config_wrapper {
+
+    config_wrapper(eosio::name self) : _self(self), _config(self, self.value) {
+        _exists = _config.exists();
```

Comment:
Maybe it isn't a big deal. But this does mean we lookup the config on all actions even if we don't need it.

Alternatively, you could have made `_cached_config` an optional and have a helper function to ensure it is present prior to any accessor or mutator of a field involving that singleton. That way DB lookups are delayed to the first time they are actually needed.

Reply:
Never mind for now. Every action has to do a lookup of the config singleton anyway. So we don't get an advantage by being smarter yet. In the future if this wrapper manages multiple singletons we will have to be smarter about caching then.